src/HealthChecks.System/FolderHealthCheck.cs

```diff
@@ -0,0 +1,41 @@
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace HealthChecks.System
+{
+    public class FolderHealthCheck : IHealthCheck
+    {
+        private readonly FolderHealthCheckOptions _folderOptions;
+
+        public FolderHealthCheck(FolderHealthCheckOptions folderOptions)
+        {
+            _folderOptions = folderOptions ?? throw new ArgumentNullException(nameof(folderOptions));
+        }
+
+        public Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                List<string>? errorList = null;
+                if (!string.IsNullOrEmpty(_folderOptions.Folder))
```

Comment:
So check returns healthy for not specified folder in options. Is this normal?

Reply:
I think it would be a good idea. If the folder list is populated dynamically and one environment doesn't use any, it would cause it to return unhealthy. This just simplifies the development effort. I'm not tied to this, but that was my thinking.

Reply:
ok